### PR TITLE
[MM-14093] Rename 'team' to 'server' and 'tab' to 'view' in most cases, some additional cleanup

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
This is the long-awaited change to update all references to 'team' to be 'server' to avoid confusion in the code. I've also change the word 'tab' to 'view' as well to more accurate portray what the view is.

There are some notes on these changes:
- `teams` and `tabs` are still identifiers used by the local configuration file, so I've left those as-is. They will get changed in a future PR when we update the config definition.
- For all references to tabs with respect to ordering and setting the active/last active one, I've left those alone as well as those will eventually be treated as a type of view, and should respect the fact that they are still tabs within the Main Window.

#### Ticket Link
Closes https://github.com/mattermost/desktop/issues/623
https://mattermost.atlassian.net/browse/MM-14093

```release-note
NONE
```

### CODE REVIEW

File: src/main/trustedOrigins.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export class TrustedOriginsStore {
     };
 
     // if permissions or targetUrl are invalid, this function will throw an error
-    // this function stablishes all the permissions at once, overwriting whatever was before
+    // this function sviewlishes all the permissions at once, overwriting whatever was before
```
Comment (enahum) [inline src/main/trustedOrigins.ts:64]: nice new word

File: src/renderer/modals/editServer/editServer.tsx [hunk 2]
```diff
@@ -7,49 +7,49 @@ import 'renderer/css/modals.css';
 import React, {useEffect, useState} from 'react';
 import ReactDOM from 'react-dom';
 
-import {MattermostTeam} from 'types/config';
+import {UniqueServer} from 'types/config';
 
 import IntlProvider from 'renderer/intl_provider';
 
-import NewTeamModal from '../../components/NewTeamModal'; //'./addServer.jsx';
+import NewServerModal from '../../components/NewServerModal'; //'./addServer.jsx';
```
Comment (enahum) [inline src/renderer/modals/editServer/editServer.tsx:14]: what does the comment mean?

File: src/renderer/modals/newServer/newServer.tsx [hunk 3]
```diff
@@ -7,11 +7,11 @@ import 'renderer/css/modals.css';
 import React, {useEffect, useState} from 'react';
 import ReactDOM from 'react-dom';
 
-import {MattermostTeam} from 'types/config';
+import {UniqueServer} from 'types/config';
 
 import IntlProvider from 'renderer/intl_provider';
 
-import NewTeamModal from '../../components/NewTeamModal'; //'./addServer.jsx';
+import NewServerModal from '../../components/NewServerModal'; //'./addServer.jsx';
```
Comment (enahum) [inline src/renderer/modals/newServer/newServer.tsx:14]: same question about the comment

**Review by enahum — APPROVED** [review]
overall lgtm, left a few comments

**Review by enahum — APPROVED** [review]

File: src/main/trustedOrigins.ts (hunk 1 above)
Comment (devinbinnie) [reply src/main/trustedOrigins.ts:64]: Whoops missed one :rofl:

File: src/common/config/index.ts [hunk 4]
```diff
@@ -36,7 +36,7 @@ export class Config extends EventEmitter {
     private appPath?: string;
 
     private registryConfig: RegistryConfig;
-    private predefinedServers: ConfigServer[];
+    private pPredefinedServers: ConfigServer[];
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/common/config/index.ts:39]: I also prefer that, but I was getting blocked by our ESLint config. In order to properly turn it on we have to upgrade to v8 (which I haven't done yet, should probably do that)

For now I just changed the rule to a warning so we can do this.

File: i18n/de.json [hunk 5]
```diff
@@ -201,18 +201,18 @@
...
+    "renderer.components.newServerModal.serverURL.description": "Die URL deines Mattermost Servers. Muss mit http:// oder https:// beginnen.",
+    "renderer.components.newServerModal.serverURL": "Server URL",
+    "renderer.components.newServerModal.serverDisplayName.description": "Der Name des Server, der auf der Desktop App Tab-Leiste angezeigt wird.",
+    "renderer.components.newServerModal.serverDisplayName": "Server-Anzeigename",
+    "renderer.components.newServerModal.error.urlRequired": "URL wird benötigt.",
+    "renderer.components.newServerModal.error.urlNeedsHttp": "URL sollte mit http:// oder https:// beginnen.",
+    "renderer.components.newServerModal.error.urlIncorrectFormatting": "Die URL ist nicht korrekt formatiert.",
+    "renderer.components.newServerModal.error.serverUrlExists": "Ein Server mit der gleichen URL existiert schon.",
+    "renderer.components.newServerModal.error.serverNameExists": "Ein Server mit diesem Namen existiert schon.",
+    "renderer.components.newServerModal.error.nameRequired": "Name wird benötigt.",
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply i18n/de.json:215]: Honestly don't know, should it? If I change it here there will be nightlies without proper translations. Not sure if that matters to us or not.

File: src/renderer/modals/editServer/editServer.tsx (hunk 2 above)
Comment (devinbinnie) [reply src/renderer/modals/editServer/editServer.tsx:14]: Ahh those are old. I'll remove them.

**Review by m1lt0n — APPROVED** [review]

**Pull request merged**